### PR TITLE
Support EditorConfig for online browser/editor.

### PR DESCRIPTION
**Pull request by kounoike**
This PR add [EditorConfig](http://editorconfig.org/) support for online browser/editor. It reads

- `indent_style`
- `indent_size` (`tab_width`)
- `end_of_line`

from `.editorconfig` file. and reflects to online browser/editor.

### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [x] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [ ] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

### CODE REVIEW

**Comment by kounoike** [comment]
Sorry, I forgot to care parse error in `.editorconfig`. I will rewrite.

File: src/main/java/editorconfig/JGitResource.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.TreeWalk;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+
+public class JGitResource implements Resource {
```
Comment (takezoe) [inline src/main/java/editorconfig/JGitResource.java]: `JGitResource` and `JGitResourcePath` can be implemented with Scala. How about implementing them as inner classes of `EditorConfigUtil`?

File: src/main/java/editorconfig/JGitResource.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    private final Repository repo;
+    private final String revStr;
+
+    Ec4jPath path;
```
Comment (takezoe) [inline src/main/java/editorconfig/JGitResource.java]: Is this field needed to be package private?

File: src/main/scala/gitbucket/core/util/EditorConfigUtil.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.ec4j.core.model.PropertyType.{EndOfLineValue, IndentStyleValue}
+import org.ec4j.core.model.{PropertyType, Version}
+import org.ec4j.core.parser.ParseException
+import org.ec4j.core.{EditorConfigConstants, EditorConfigLoader, ResourceProperties, ResourcePropertiesService}
+import org.eclipse.jgit.api.Git
+
+object EditorConfigUtil {
+  val TabSizeDefault: Int = 8
+  val NewLineModeDefault: String = "auto"
+  val UseSoftTabsDefault = false
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/util/EditorConfigUtil.scala]: These static variables are able to be private.

**Comment by kounoike** [comment]
@takezoe OK, I updated this PR.

**Comment by kounoike** [comment]
@takezoe Thanks for merging!

In editorconfig/editorconfig-core-java#14 @ppalaga says

> Could you please add GitBucket to "who is using ec4j" section of our README

Could you reply it?

**Comment by takezoe** [comment]
Sure. I replied.

**Pull request merged**